### PR TITLE
Add districts and type to the users query on graphql

### PR DESCRIPTION
**Pull request by beligante**

### CODE REVIEW

File: apps/re_web/test/graphql/accounts/query_test.exs [hunk 1]
```diff
@@ -22,6 +22,8 @@ defmodule ReWeb.GraphQL.Accounts.QueryTest do
   end
 
   describe "userProfile" do
+
+    @tag dev: true
```
Comment (rhnonose) [inline apps/re_web/test/graphql/accounts/query_test.exs]: whack

File: apps/re_web/lib/graphql/resolvers/addresses.ex [hunk 2]
```diff
@@ -40,6 +41,17 @@ defmodule ReWeb.Resolvers.Addresses do
 
   def districts(_, _), do: {:ok, Neighborhoods.districts()}
 
+  def districts_by_user(_, %{context: %{current_user: current_user}}) do
+    districts =
+      current_user
+      |> Repo.preload(:districts)
+      |> Map.get(:districts)
+
+    IO.inspect(districts)
```
Comment (rhnonose) [inline apps/re_web/lib/graphql/resolvers/addresses.ex]: whack

File: apps/re_web/lib/graphql/resolvers/accounts.ex [hunk 3]
```diff
@@ -34,8 +35,10 @@ defmodule ReWeb.Resolvers.Accounts do
   def profile(params, %{context: %{current_user: current_user}}) do
     with {:ok, id} <- get_user_id(params, current_user),
          {:ok, user} <- Users.get(id),
-         :ok <- Bodyguard.permit(Users, :show_profile, current_user, user),
-         do: {:ok, user}
+         :ok <- Bodyguard.permit(Users, :show_profile, current_user, user) do
+      Repo.preload(user, :districts)
```
Comment (rhnonose) [inline apps/re_web/lib/graphql/resolvers/accounts.ex]: We don't use preload with graphql attributes, see examples using dataloader.

File: apps/re_web/lib/graphql/resolvers/addresses.ex [hunk 4]
```diff
@@ -40,6 +41,17 @@ defmodule ReWeb.Resolvers.Addresses do
 
   def districts(_, _), do: {:ok, Neighborhoods.districts()}
 
+  def districts_by_user(_, %{context: %{current_user: current_user}}) do
+    districts =
+      current_user
+      |> Repo.preload(:districts)
```
Comment (rhnonose) [inline apps/re_web/lib/graphql/resolvers/addresses.ex]: We don't use preload with graphql attributes, see examples using dataloader.

File: apps/re_web/lib/graphql/resolvers/addresses.ex [hunk 5]
```diff
@@ -40,6 +41,17 @@ defmodule ReWeb.Resolvers.Addresses do
 
   def districts(_, _), do: {:ok, Neighborhoods.districts()}
 
+  def districts_by_user(_, %{context: %{current_user: current_user}}) do
```
Comment (pmargreff) [inline apps/re_web/lib/graphql/resolvers/addresses.ex]: This name looks right, but something like `districts_by_broker` or `districts_for_broker_users` make the intention.

File: apps/re/lib/addresses/districts.ex [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  def data(params), do: Dataloader.Ecto.new(Re.Repo, query: &query/2, default_params: params)
+
+  def query(query, _args), do: query
+
+  def districts_by_broker_user(user_uuid) do
+    Repo.all(
+      from(
+        d in District,
+        join: bd in BrokerDistrict,
```
Comment (rhnonose) [inline apps/re/lib/addresses/districts.ex]: No joins when using dataloader either.

**Pull request merged**